extension/src/config/config.js

```diff
@@ -1,58 +1,87 @@
+        ctpProjectKey
+      )}`
+    )
   return {
-    projectKey: process.env.CTP_PROJECT_KEY,
-    clientId: process.env.CTP_CLIENT_ID,
-    clientSecret: process.env.CTP_CLIENT_SECRET,
-    apiUrl:
-      process.env.CTP_HOST || 'https://api.europe-west1.gcp.commercetools.com',
+    clientId: ctpConfig.clientId,
+    clientSecret: ctpConfig.clientSecret,
+    projectKey: ctpProjectKey,
+    apiUrl: ctpConfig.apiUrl || 'https://api.europe-west1.gcp.commercetools.com',
     authUrl:
-      process.env.CTP_AUTH_URL ||
-      'https://auth.europe-west1.gcp.commercetools.com',
+      ctpConfig.authUrl || 'https://auth.europe-west1.gcp.commercetools.com',
+    ensureResources: config.ensureResources !== 'false',
   }
 }
 
-function getAdyenCredentials() {
+function getAdyenConfig(adyenMerchantAccount) {
+  const adyenConfig = config.adyen[adyenMerchantAccount]
+  if (!adyenConfig)
+    throw new Error(
+      `Configuration for adyenMerchantAccount is not provided. Please update the configuration: ${JSON.stringify(
+        adyenMerchantAccount
+      )}`
+    )
   return {
-    merchantAccount: process.env.ADYEN_MERCHANT_ACCOUNT,
-    apiKey: process.env.ADYEN_API_KEY,
-    apiBaseUrl:
-      process.env.ADYEN_API_BASE_URL || 'https://checkout-test.adyen.com/v52',
+    apiKey: adyenConfig.apiKey,
+    apiBaseUrl: adyenConfig.apiBaseUrl || 'https://checkout-test.adyen.com/v52',
     legacyApiBaseUrl:
-      process.env.ADYEN_LEGACY_API_BASE_URL ||
+      adyenConfig.legacyApiBaseUrl ||
       'https://pal-test.adyen.com/pal/servlet/Payment/v52',
-    clientKey: process.env.ADYEN_CLIENT_KEY || '',
+    clientKey: adyenConfig.clientKey || '',
   }
 }
 
-module.exports.load = () => {
-  /**
-   * Load configuration from several sources in this order (last has highest priority):
-   * - default config
-   * - file config
-   * - ctp credentials from env variables
-   * - ctp config
-   * - env config
-   */
-
-  const config = _.merge(
-    getEnvConfig(),
-    { ctp: getCTPEnvCredentials() },
-    { adyen: getAdyenCredentials() }
-  )
-
-  return config
+function getAllCtpProjectKeys() {
+  return Object.keys(config.commercetools)
+}
+
+function getAllAdyenMerchantAccounts() {
+  return Object.keys(config.adyen)
+}
+
+function loadAndValidateConfig() {
+  try {
+    config = JSON.parse(process.env.ADYEN_INTEGRATION_CONFIG)
+  } catch (e) {
+    throw new Error(
+      'Adyen integration configuration is not provided in the JSON format'
+    )
+  }
+  const numberOfCtpConfigs = Object.keys(config.commercetools).length
+  const numberOfAdyenConfigs = Object.keys(config.adyen).length
+  if (numberOfCtpConfigs === 0)
+    throw new Error(
+      'Please add at least one commercetools project to the config'
+    )
+  if (numberOfAdyenConfigs === 0)
+    throw new Error(
+      'Please add at least one Adyen merchant account to the config'
+    )
+}
+
+loadAndValidateConfig()
```

Comment:
not sure from calling on global scope, should we call it instead of a global call? Are you considering a serverless way 🤔

Reply:
This will be called anytime this file is being used in the code. So it's sth like a necessary condition for this file to work. It fails really fast if sth is wrong. I don't see any problems here with serverless, what is your concern with serverless?

Reply:
You seem correct, should not be an issue 🤔 👍🏽